packages/@aws-cdk/aws-ecs-patterns/lib/base/application-multiple-target-groups-service-base.ts

```diff
@@ -0,0 +1,502 @@
+   *
+   * @default false
+   */
+  readonly enableECSManagedTags?: boolean;
+
+  /**
+   * The options for configuring an Amazon ECS service to use service discovery.
+   *
+   * @default - AWS Cloud Map service discovery is not enabled.
+   */
+  readonly cloudMapOptions?: CloudMapOptions;
+
+  /**
+   * Properties to specify ECS target groups. At least one target group should be specified.
+   *
+   * @default - default portMapping registered as
+   * target group and attached to the first defined listener
+   */
+  readonly targetGroups?: ApplicationTargetProps[];
+}
+
+export interface ApplicationTargetProps {
+  /**
+   * The port number of the container. Only applicable when using application/network load balancers.
+   */
+  readonly containerPort: number;
+
+  /**
+   * The protocol used for the port mapping. Only applicable when using application load balancers.
+   *
+   * @default ecs.Protocol.TCP
+   */
+  readonly protocol?: Protocol;
+
+  /**
+   * Name of the listener the target group attached to.
+   *
+   * @default - default listener (first added listener)
+   */
+  readonly listener?: string;
+
+  /**
+   * Priority of this target group
+   *
+   * The rule with the lowest priority will be used for every request.
+   * If priority is not given, these target groups will be added as
+   * defaults, and must not have conditions.
+   *
+   * Priorities must be unique.
+   *
+   * @default Target groups are used as defaults
+   */
+  readonly priority?: number;
+
+  /**
+   * Rule applies if the requested host matches the indicated host
+   *
+   * May contain up to three '*' wildcards.
+   *
+   * Requires that priority is set.
+   *
+   * @see https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-listeners.html#host-conditions
+   *
+   * @default No host condition
+   */
+  readonly hostHeader?: string;
+
+  /**
+   * Rule applies if the requested path matches the given path pattern
+   *
+   * May contain up to three '*' wildcards.
+   *
+   * Requires that priority is set.
+   *
+   * @see https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-listeners.html#path-conditions
+   *
+   * @default No path condition
+   */
+  readonly pathPattern?: string;
+}
+
+export interface ApplicationLoadBalancerProps {
+  /**
+   * Name of the load balancer (required if loadBalancer is not specified)
+   *
+   * Note that this field will be ignored if loadBalancer is set.
+   */
+  readonly name?: string;
+
+  /**
+   * Listeners (at least one listener) that attached to this load balancer (required if loadBalancer is not specified)
+   *
+   * Note that this field will be ignored if loadBalancer is set.
+   *
+   * @default - none
+   */
+  readonly listeners?: ApplicationListenerProps[];
+
+  /**
+   * Determines whether the Load Balancer will be internet-facing.
+   *
+   * Note that this field will be ignored if loadBalancer is set.
```

Comment:
this should be loadBalancers yes?

Reply:
Oh this is no typo because for each loadbalancer you can define it outside and pass in as loadBalancer and if loadBalancer this property is defined, publicLoadBalancer et al will be meaningless because those properties are used to create a load balancer.